### PR TITLE
Made the explanation of the fee calculation clearer

### PR DESCRIPTION
**Pull request by qbzzt**
Closing: INT-324

Added an example to make the fee calculation clearer

### CODE REVIEW

**Comment by Nicca42** [comment]
If you can resolve the conflicts here I will approve/ merge it in 🙏 @qbzzt

**Comment by qbzzt** [comment]
Conflict resolved. This is so much easier than in the Middle East ;-)

File: src/docs/developers/l2/new-fees.md [hunk 1]
```diff
@@ -41,10 +69,10 @@ const l1FeeInWei = await OVM_GasPriceOracle.getL1Fee(serializedTx)
 - You can show the math :
```
[the remark this answers is not in the document]
Comment (qbzzt) [reply src/docs/developers/l2/new-fees.md:69]: I don't think "but would have overpaid" works great either. How about:

> If they increase it, they will get their transaction included immediately (same as with the correct price) but at a higher cost

File: src/docs/developers/l2/new-fees.md [hunk 2]
```diff
@@ -5,34 +5,62 @@ lang: en-US
...
+   The exact function is `l1GasPrice * 1.5 * (2750 + calldata gas)`. 
+
+
+For example, lets look at [this transaction](https://kovan-optimistic.etherscan.io/tx/0xcf2e2f7f7088e4332a2b5369f85b1bafa8a8a007122e228d90778a1d14c41286). The **L2 execution fee** is the gas used, `195,057`, times the current l2 gas price, which is 0.001 gwei. In other words, approximately 195 gwei. The **L1 security fee**, on the other hand, is based on a gas usage of of `1.5 * (2750 + 16*11 + 4*25)` (bytes with zero only cost four gas), or 4539 gas. At the current L1 gas price of 150 gwei, this costs 680,850 gwei, about 3500 times the 
+**L2 execution fee**.
+
+::: tip 
+This transaction is typical. In almost all cases **L2 execution fee** is negligible compared to the **L1 security fee**. 
+
+Because the **L1 security fee** is the major cost of transactions, it is critical for wallets to display the **L1 security fee** to the user before the user approves the transaction.
```
[the remark this answers is not in the document]
Comment (qbzzt) [reply src/docs/developers/l2/new-fees.md:32]: This doc was originally written by Kevin, and he included a lot of design advice, such as 

> We recommend displaying an estimated fee to users via the following math:

and

> We recommend building error handling around the `Fee too Low` error detailed below, to allow users to re-calculate their `tx.gasPrice` and resend their transaction if fees spike.

I was following that practice. Maybe we should discuss tomorrow whether our docs should include design advice or not.

File: src/docs/developers/l2/new-fees.md [hunk 3]
```diff
@@ -5,34 +5,62 @@ lang: en-US
...
+
+## For backend developers
 - You must send your transaction with a tx.gasPrice that is greater than or equal to the sequencer's l2 gas price. You can read this value from the Sequencer by querying the `OVM_GasPriceOracle` contract  (`OVM_GasPriceOracle.gasPrice`) or by simply making an RPC query to `eth_gasPrice`.  If you don't specify your `gasPrice` as an override when sending a transaction , `ethers` by default queries `eth_gasPrice` which will return the lowest acceptable L2 gas price.
 - You can set your `tx.gasLimit` however you might normally set it (e.g. via `eth_estimateGas`). You can expect that gas usage for transactions on Optimism Ethereum will be identical to gas usage on Ethereum.
 - We recommend building error handling around the `Fee too Low` error detailed below, to allow users to re-calculate their `tx.gasPrice` and resend their transaction if fees spike.
 
-## For Frontend and Wallet developers:
+## For Frontend and Wallet developers
 - We recommend displaying an estimated fee to users via the following math:
    1. To estimate the L1 (security) fee that the user should expect to pay. For example, calculating the L1 fee for sending a USDC transfer:
```
[the remark this answers is not in the document]
Comment (qbzzt) [reply src/docs/developers/l2/new-fees.md:45]: Thank you for catching that.

File: src/docs/developers/l2/new-fees.md [hunk 4]
```diff
@@ -5,34 +5,62 @@ lang: en-US
 
 # {{ $frontmatter.title }}
 
+## The logic of Optimistic Ethereum fees
 You can see how the fee is calculated and deducted [here](../../users/fees-2.0.md).
 
-## For backend developers:
+Every Optimistic Ethereum transaction has two costs:
```
[the remark this answers is not in the document]
Comment (qbzzt) [reply src/docs/developers/l2/new-fees.md:11]: But there is additional information which is more details than we need in a user doc. To present it without the short explanations from the user docs would make for a choppy, unclear document.

File: src/docs/developers/l2/new-fees.md [hunk 5]
```diff
@@ -5,34 +5,62 @@ lang: en-US
...
+
+   This is cost is accessible via the normal Ethereum mechanisms, you can estimate it using ethers' [`estimateGas`](https://docs.ethers.io/v5/api/contract/contract/#contract-estimateGas), for example.
+
+1. The **L1 security fee** pays for the cost of publishing the transaction on L1 (the cost of Ethereum equivalent security). It is deducted automatically from the user's ETH balance on Optimistic Ethereum. It is based on two factors:
+
+   - `l1GasPrice` is the current gas price for L1 transactions. You can see it [here](https://public-grafana.optimism.io/d/9hkhMxn7z/public-dashboard?orgId=1&refresh=5m).
+
+   - The transaction's call data, because that call data has to be written on L1 as part of publishing the transaction. [Call data costs approximately 16 gas per byte](https://eips.ethereum.org/EIPS/eip-2028).
+
```
[the remark this answers is not in the document]
Comment (elenadimitrova) [reply src/docs/developers/l2/new-fees.md:22]: This is not resolved

File: src/docs/developers/l2/new-fees.md [hunk 6]
```diff
@@ -5,34 +5,62 @@ lang: en-US
...
+   The exact function is `l1GasPrice * 1.5 * (2750 + calldata gas)`. 
+
+
+For example, lets look at [this transaction](https://kovan-optimistic.etherscan.io/tx/0xcf2e2f7f7088e4332a2b5369f85b1bafa8a8a007122e228d90778a1d14c41286). The **L2 execution fee** is the gas used, `195,057`, times the current l2 gas price, which is 0.001 gwei. In other words, approximately 195 gwei. The **L1 security fee**, on the other hand, is based on a gas usage of of `1.5 * (2750 + 16*11 + 4*25)` (bytes with zero only cost four gas), or 4539 gas. At the current L1 gas price of 150 gwei, this costs 680,850 gwei, about 3500 times the
```
[the remark this answers is not in the document]
Comment (elenadimitrova) [reply src/docs/developers/l2/new-fees.md:26]: This is also not resolved.

File: src/docs/developers/l2/new-fees.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (elenadimitrova) [reply src/docs/developers/l2/new-fees.md:32]: Following what we discussed on the call, this maybe relevant advice for the new section on Guides for wallet developers. Will make a note to move it there in the Doc upgrade project. Resolving for now.

File: src/docs/developers/l2/new-fees.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (elenadimitrova) [reply src/docs/developers/l2/new-fees.md:11]: I'm making a note to deduplicate this better in the Doc Upgrade work.

**Review by elenadimitrova — APPROVED** [review]

**Review by Nicca42 — APPROVED** [review]

**Pull request merged**